cpp/src/reductions/scan/scan_inclusive.cu

```diff
@@ -330,6 +330,270 @@ std::unique_ptr<column> inclusive_rank_scan(column_view const& order_by,
+{
+  rmm::device_uvector<thrust::pair<double, double>> pairs(input.size(), stream, mr);
+
+  thrust::transform(rmm::exec_policy(stream),
+                    input.begin<double>(),
+                    input.end<double>(),
+                    pairs.begin(),
+                    [=] __host__ __device__(double input) -> thrust::pair<double, double> {
+                      return thrust::pair<double, double>(beta, (1.0 - beta) * input);
+                    });
+
+  // TODO: the first pair is WRONG using the above. Reset just that pair
+
+  thrust::transform(rmm::exec_policy(stream),
+                    input.begin<double>(),
+                    input.begin<double>() + 1,
+                    pairs.begin(),
+                    [=] __host__ __device__(double input) -> thrust::pair<double, double> {
+                      return thrust::pair<double, double>(beta, input);
+                    });
+
+  compute_recurrence(pairs, stream);
+  // copy the second elements to the output for now
+  thrust::transform(rmm::exec_policy(stream),
+                    pairs.begin(),
+                    pairs.end(),
+                    output.begin<double>(),
+                    [=] __host__ __device__(thrust::pair<double, double> pair) -> double {
+                      return thrust::get<1>(pair);
+                    });
+}
+
+std::unique_ptr<column> ewma(column_view const& input,
+                             double com,
+                             bool adjust,
+                             rmm::cuda_stream_view stream,
+                             rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(cudf::is_floating_point(input.type()), "Column must be floating point type");
+
+  double beta = 1.0 - (1.0 / (com + 1.0));
+
+  auto output = make_fixed_width_column(cudf::data_type{cudf::type_id::FLOAT64}, input.size());
+  mutable_column_view output_mutable_view = output->mutable_view();
+
+  if (adjust) {
+    compute_ewma_adjust(input, beta, output_mutable_view, stream, mr);
+  } else {
+    compute_ewma_noadjust(input, beta, output_mutable_view, stream, mr);
+  }
+  return output;
+}
+
+std::unique_ptr<column> ewmvar(column_view const& input,
+                               double com,
+                               bool adjust,
+                               rmm::cuda_stream_view stream,
+                               rmm::mr::device_memory_resource* mr)
+{
+  // get means
+  std::unique_ptr<column> means = ewma(input, com, adjust, stream, mr);
+  auto means_view               = means.get()[0].mutable_view();
+
+  // construct deltas: (x_i - mu_i)(x_i - mu_{i-1})
+  rmm::device_uvector<double> d_this(input.size(), stream, mr);
+  rmm::device_uvector<double> d_last(input.size(), stream, mr);
+
+  // x_i - mu_i
+  thrust::transform(rmm::exec_policy(stream),
+                    input.begin<double>() + 1,
+                    input.end<double>(),
+                    means_view.begin<double>() + 1,
+                    d_this.begin() + 1,
```

Comment:
Note you are not writing to `d_this[0]` here, and since it is a `device_uvector`, it is uninitialized.